### DISCUSSION TITLE
Option not to recurse

### DISCUSSION THREAD

**Discussion by mrx23dot** [manusimidt/py-xbrl#34] [Ideas]
Is there an option not to recurse for other documents?
95% of the data is in the main document, and downloading 24 extra files is a lot.

**Comment by mrx23dot** [comment]
As I see the only problem in the main file is that one fact can have multiple entries, and we don't know which one is the latest.
Do we need every other file for that to resolve it?

**Comment by manusimidt** [comment]
Currently there is no option to disable the parsing of documents that are imported by the instance document. The goal of this parser is to parse all information contained in the XBRL filing. A single XBRL Filing not only transfers individual facts, but also to defines multidimensional information, mathematical relationships between concepts and their presentation in the report.

Normally you don't have to download 24 additional files each time. Many of the downloaded files are taxonomies which are used by many submissions. These must then be downloaded only once.

If you for example parse the following two submissions:
`https://www.sec.gov/Archives/edgar/data/789019/000156459017014900/msft-20170630.xml` and `https://www.sec.gov/Archives/edgar/data/789019/000156459017020171/msft-20170930.xml` the parser will download all the standard EDGAR taxonomies (dei, us-gaap, e.t.c) only once for the first submission and stores them into the defined cache destination.
When parsing the second submission the parser automatically detects that those taxonomies where already downloaded and loads them directly from the cache without making a request.
